### PR TITLE
Effect shouldn't affect non-used properties

### PR DESCRIPTION
**Pull request by spydon**
# Description

Right now an effect resets all the properties of a position component when it is done, it should only reset the once it has affected.
For example a ScaleEffect will reset the position of the PositionComponent when it is done.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on the latest `master`
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: lib/effects/effects.dart [hunk 1]
```diff
@@ -157,22 +157,31 @@ abstract class PositionComponentEffect
...
-  void setComponentToOriginalState() {
+  /// Only change the parts of the component that is affected by the
+  /// effect, and only set the state if it is the root effect (not part of
+  /// another effect, like children of a CombinedEffect or SequenceEffect).
+  void _setComponentState(Vector2 position, double angle, Vector2 size) {
     if (isRootEffect()) {
-      component?.position?.setFrom(originalPosition);
-      component?.angle = originalAngle;
-      component?.size?.setFrom(endSize);
+      if (originalPosition != endPosition) {
```
Comment (erickzanardo) [inline lib/effects/effects.dart:165]: This if shouldn't include the `position` variable?

File: lib/effects/effects.dart [hunk 2]
```diff
@@ -157,22 +157,31 @@ abstract class PositionComponentEffect
...
+        component?.position?.setFrom(position);
+      }
+      if (originalAngle != endAngle) {
```
Comment (erickzanardo) [inline lib/effects/effects.dart:168]: Same question here

File: lib/effects/effects.dart [hunk 3]
```diff
@@ -157,22 +157,31 @@ abstract class PositionComponentEffect
...
+        component?.angle = angle;
+      }
+      if (originalSize != originalSize) {
```
Comment (erickzanardo) [inline lib/effects/effects.dart:171]: This if will never be false :P

File: lib/effects/effects.dart (hunk 1 above)
Comment (spydon) [reply lib/effects/effects.dart:165]: No, it is just checking whether this effect made any changes on the component.
The position variable will be either `endPosition` or `originalPosition`, but this way I don't have to duplicate the code.

File: lib/effects/effects.dart (hunk 3 above)
Comment (spydon) [reply lib/effects/effects.dart:171]: Good catch, should be `endSize`!

**Review by erickzanardo — APPROVED** [review]
LGTM

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**